BeeKit/UI/BSButton.swift

```diff
@@ -13,23 +13,52 @@ public class BSButton : UIButton {
     
     required init?(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
+        
+        registerForTraitChanges(
+            [UITraitUserInterfaceLevel.self]) {
+                (self: Self, previousTraitCollection: UITraitCollection) in
+                self.resetStyle()
+            }
+        
         self.setUp()
     }
     
     override init(frame: CGRect) {
         super.init(frame: frame)
+        
+        registerForTraitChanges(
+            [UITraitUserInterfaceLevel.self]) {
+                (self: Self, previousTraitCollection: UITraitCollection) in
+                self.resetStyle()
+            }
+        
         self.setUp()
     }
     
     func setUp() {
         self.titleLabel?.font = UIFont.beeminder.defaultBoldFont
         self.setTitleColor(UIColor.Beeminder.yellow, for: UIControl.State())
-        self.tintColor = .black
+        self.tintColor = dynamicTintFillColor
         self.configuration = .filled()
         
         self.layer.borderColor = UIColor.Beeminder.yellow.cgColor
         self.layer.borderWidth = 1
         self.layer.cornerRadius = 4
     }
     
+    private func resetStyle() {
+        self.tintColor = dynamicTintFillColor
+    }
+    
+    private let dynamicTintFillColor = UIColor { traitCollection in
+        switch traitCollection.userInterfaceStyle {
+        case .dark:
+            return UIColor.black
+        default:
+            return UIColor(red: 59.0/255.0,
```

Comment:
I wonder what it would look like if we used the same grey as the +/- buttons. Possibly too low contrast with the yellow?

Reply:
I think you are right. The contrast seems to work for me on the sign in button but looks too low contrast elsewhere.

How about something like this - yellow on black in dark and black on grey in light mode?

https://github.com/user-attachments/assets/34a7d634-3826-45d5-ab82-a02c95e37981

Reply:
What if we skipped the border entirely and just had the solid button, either gray or yellow?

Reply:
Grey (OK):
![Screenshot 2024-10-06 at 2 49 29 PM](https://github.com/user-attachments/assets/2159e1fd-95c4-4e18-a4c7-d22800a18eda)


Yellow + Black Text (Probably too loud):
![Screenshot 2024-10-06 at 2 50 42 PM](https://github.com/user-attachments/assets/e52f87eb-18e6-4bb6-9705-01f826faef1c)

Yellow + White Text (Unreadable):
![Screenshot 2024-10-06 at 2 51 14 PM](https://github.com/user-attachments/assets/95ecf4d4-25d6-40a8-8523-183f0716541f)

Reply:
And for completeness, default system blue:
![Screenshot 2024-10-06 at 2 53 23 PM](https://github.com/user-attachments/assets/1ec43566-bfad-46b8-9910-067b583a6e94)

Reply:
And the current target? Yellow text on Black background with yellow border in dark mode and black text on grey and without border in light mode?

Reply:
My take would be:
* Yellow border and text, black background when shown in dark mode
* Black on grey without border in light mode

But I don't feel like I have particularly good graphics design taste, so I'm happy for you to make a call on what you want to try, and we can put it out to test flight to get wider feedback.